confgenerator/resourcedetector/detector.go

```diff
@@ -0,0 +1,33 @@
+package resourcedetector
+
+import (
+	gcp_metadata "cloud.google.com/go/compute/metadata"
+)
+
+type Detector interface {
+	GetType() string
+}
+
+// Get a Detector for the current environment, which can be used
+// to get all the attributes
+func GetDetector() (Detector, error) {
+	switch {
+	case gcp_metadata.OnGCE():
+		return GetGCEDetector()
+	default:
+		return GetEmptyDetector()
+	}
+}
+
+// EmptyDetector that returns an empty detector without any attributes
+// for non-recognized environments
+type EmptyDetector struct {
```

Comment:
Not sure if it would be better to call this `UnrecognizedPlatformDetector` or something similar.

Reply:
+1 for `UnrecognizedPlatformDetector`.